components/ipfs/ipfs_utils.cc

```diff
@@ -134,17 +135,25 @@ bool IsIpfsDisabledByPolicy(PrefService* prefs) {
          !prefs->GetBoolean(kIPFSEnabled);
 }
 
+bool HasIPFSOrIPNSPath(const GURL& gurl) {
+  return gurl.is_valid() && (HasIPFSPath(gurl) || HasIPNSPath(gurl));
+}
+
 bool HasIPFSPath(const GURL& gurl) {
   const auto& path = gurl.path();
-  return gurl.is_valid() && ((path.find("/ipfs/") != std::string::npos) ||
-                             (path.find("/ipns/") != std::string::npos));
+  return gurl.is_valid() && base::Contains(path, "/ipfs/");
+}
+
+bool HasIPNSPath(const GURL& gurl) {
+  const auto& path = gurl.path();
+  return gurl.is_valid() && base::Contains(path, "/ipns/");
 }
 
 bool IsDefaultGatewayURL(const GURL& url, PrefService* prefs) {
   DCHECK(prefs);
   std::string gateway_host = GetDefaultIPFSGateway(prefs).host();
   return url.DomainIs(gateway_host) &&
-         (HasIPFSPath(url) ||
+         (HasIPFSOrIPNSPath(url) ||
           url.DomainIs(std::string("ipfs.") + gateway_host) ||
```

Comment:
these may also be wrong cc @pes10k because I can't remember what the gateway did and I also thought we were getting rid of the default gateway as part of this, but still allowing people to enter one manually. @diracdeltas ?